index.html

```diff
@@ -347,23 +347,24 @@ <h2>The <dfn>ThingDescription</dfn> type</h2>
         Therefore, this API expects the {{ThingDescription}} objects be validated before used as parameters. This specification defines a basic <a>TD</a> validation as follows.
       </p>
       <div>
-        To <dfn>validate a TD</dfn> given |td:ThingDescription|, run the following steps:
+        run the following steps:
         <ol>
           <li>
             If |td| is not an object, [= exception/throw =] a
             {{"TypeError"}} and abort these steps.
           </li>
+          <li>
+            If <a href="https://www.w3.org/TR/2019/CR-wot-thing-description-20190516/#json-schema-for-validation">JSON schema
+              validation</a> fails on |td|, [= exception/throw =] a
+            {{"TypeError"}} and abort these steps.
+          </li>
           <li>
             If any of the mandatory properties defined in [[!WOT-TD]] for
             <a href="https://www.w3.org/TR/2019/CR-wot-thing-description-20190516/#thing">Thing</a>
             that don't have <a href="https://www.w3.org/TR/2019/CR-wot-thing-description-20190516/#sec-default-values">default definitions</a>
             are missing from |td|, [= exception/throw =] a
             {{"TypeError"}} and abort these steps.
           </li>
-          <li>
-            If <a href="https://www.w3.org/TR/2019/CR-wot-thing-description-20190516/#json-schema-for-validation">JSON schema validation</a> fails on |td|, [= exception/throw =] a
-            {{"TypeError"}} and abort these steps.
-          </li>
```

Comment:
@egekorkan comment on this change:
> The JSON Schema validation should be before step 2 and that way if something fails the schema validation, it must have values for terms that do not have defaults. The schema validation does not expect a TD with defaults filled in.

Reply:
I'd leave as it was. As far as I understand if a property is marked as *required* in JSONSchema even if it has a default value, it must appear in the validating object. However, I'd add a step to make the process a little bit clearer:
1. Check if it is an Object
2. Check if any of the mandatory fields with no default are not defined.
3. Fill mandatory fields with default
4. Do JSONSchema validation

Reply:
I agree with the sequence proposed above.